content/en/blog/_posts/2018-05-25-Airflow-Kubernetes-Operator.md

```diff
@@ -0,0 +1,141 @@
+                          cmds=["python","-c"],
+                          arguments=["print('hello world')"],
+                          labels={"foo": "bar"},
+                          name="fail",
+                          task_id="failing-task",
+                          get_logs=True,
+                          dag=dag
+                          )
+
+passing.set_upstream(start)
+failing.set_upstream(start)
+```
+<img src="/images/blog/2018-05-25-Airflow-Kubernetes-Operator/2018-05-25-basic-dag-run.png" width="85%" alt="Basic DAG Run" />
+
+## But how does this relate to my workflow?
+
+While this example only uses basic images, the magic of Docker is that this same DAG will work for any image/command pairing you want. The following is a recommended CI/CD pipeline to run production-ready code on an Airflow DAG.
+
+### 1: PR in github
+Use Travis or Jenkins to run unit and integration tests, bribe your favorite team-mate into PR'ing your code, and merge to the master branch to trigger an automated CI build.
+
+### 2: CI/CD via Jenkins -> Docker Image
+
+There are a multitude on articles on [generating Docker files within a Jenkins build](https://getintodevops.com/blog/building-your-first-docker-image-with-jenkins-2-guide-for-developers). It's a good rule of thumb that you should never use a user-generated Docker image in a production build. By reserving release tags to the Jenkins user, you can ensure that malicious or untested code will never be run by your production Airflow instances.
+
+### 3: Airflow launches task 
+
+Finally, update your DAGs to reflect the new release version and you should be ready to go!
+
+```python
+production_task = KubernetesPodOperator(namespace='default',
+                          # image="my-production-job:release-1.0.1", <-- old release
+                          image="my-production-job:release-1.0.2",
+                          cmds=["python","-c"],
+                          arguments=["print('hello world')"],
+                          name="fail",
+                          task_id="failing-task",
+                          get_logs=True,
+                          dag=dag
+                          )
+```
+                          
+                          
+# So when will I be able to use this?
+
+The Kubernetes Operator and Kubernetes Executor (article to come) have both been merged into the [1.10 release branch of Airflow](https://github.com/apache/incubator-airflow/tree/v1-10-test) (the executor in experimental mode). While this branch is still in the early stages, we hope to see it released for wide release in the next few months.
+
+For those interested in getting involved with this effort, Apache Airflow has an extremely active mailing list at dev@airflow.apache.org. On the kubernetes side, discussions about Airflow on k8s (as well as Spark, HDFS, and other big data technologies) take plase on the #SIG-BIG-DATA channel on https://kubernetes.slack.com.
```

Comment:
It's a good start - you could also use something like the get-involved section at the bottom of https://databricks.com/blog/2018/03/06/apache-spark-2-3-with-native-kubernetes-support.html

Reply:
@foxish I have shamelessly plagiarized your thank you section, but you know what they say "something, something flattery" ;)